Arinthros/130 fix quoted module names

arinthros opened the pull request:
…eateStringLiteral

Attempts to resolve https://github.com/englercj/tsd-jsdoc/issues/130

This still needs testing and validation, but want to make sure the logic is sound prior to doing extra work.

englercj commented on src/create_helpers.ts:
```diff
@@ -507,9 +507,15 @@ export function createModule(doclet: INamespaceDoclet, nested: boolean, children
-        body = ts.createModuleBlock(children as ts.Statement[]);
-
-    const name = ts.createStringLiteral(doclet.name);
+				body = ts.createModuleBlock(children as ts.Statement[]);
```
Looks like there is a mix of tabs and spaces here, can you normalize that?
```suggestion
        body = ts.createModuleBlock(children as ts.Statement[]);
```

englercj commented on src/create_helpers.ts:
```diff
@@ -507,9 +507,15 @@ export function createModule(doclet: INamespaceDoclet, nested: boolean, children
-    const name = ts.createStringLiteral(doclet.name);
+				body = ts.createModuleBlock(children as ts.Statement[]);
+
+		const nameArray = [...doclet.name];
+		// Check if name is encapsulated in quotes
+		if (nameArray[0] === '"' && nameArray[nameArray.length] === '"' || nameArray[0] === "'" && nameArray[nameArray.length] === "'"){
+			nameArray.shift();
+			nameArray.pop();
+		};
+    const name = ts.createStringLiteral(nameArray.join(''));
```
This should be length - 1 to check the last element.

Also you don't have to convert it to a string to check elements:

```suggestion
    let nameStr = doclet.name;
    if ((nameStr[0] === '"' && nameStr[name.length - 1] === '"')
        || (nameStr[0] === '\'' && nameStr[name.length - 1] === '\''))
    {
        nameStr = nameStr.substr(1, name.length - 2);
    }
    const name = ts.createStringLiteral(nameStr);
```

arinthros commented:
@englercj Thanks for the review! I updated the logic and tab spacing per your comments. I also added a test for this case `module_quoted`. Let me know if there's anything else you want changed.

englercj commented on src/create_helpers.ts:
```diff
@@ -592,9 +592,13 @@ export function createModule(doclet: INamespaceDoclet, nested: boolean, children
-    const name = ts.createStringLiteral(doclet.name);
+    let nameStr = doclet.name;
+    if (nameStr[0] === '"' && nameStr[nameStr.length - 1] === '"' || nameStr[0] === '\'' && nameStr[nameStr.length - 1] === '\''){
+      nameStr = nameStr.substr(1, nameStr.length - 2);
```
This is not the same as the code I had suggested. It is missing some necessary parens and the spacing and brace format are both incorrect. I'd recommend using the suggestion I posted, or committing it directly from github.

arinthros replied:
Updated per the suggestion. Changed suggested references of `name` to `nameStr` since name is defined later.

englercj commented on test/specs/module.ts:
```diff
@@ -1,5 +1,6 @@
 suite('Module Checks', () => {
-    tsdJsdocTestCase('All', 'module_all');
+		tsdJsdocTestCase('All', 'module_all');
+		tsdJsdocTestCase('Quoted', 'module_quoted');
```
Tabs here too, should be 4 spaces.

arinthros replied:
Not sure what's going on with my IDE today. Sorry!
